src/ordered_dict.jl

```diff
@@ -311,7 +311,7 @@ function _setindex!(h::OrderedDict, v, key, index)
     sz = length(h.slots)
     cnt = nk - h.ndel
     # Rehash now if necessary
-    if h.ndel >= ((3*nk)>>2) || cnt*3 > sz*2
+    if h.ndel >= ((3*sz)>>2) || cnt*3 > sz*2
```

Comment:
Previously, when `nk` was less than 4, we would rehash every time.  This was probably unintended.

The fix here mimics `Base.Dict`, but the implementation there is different--keys are stored in an array the same shape/length as slots, whereas here, the keys are stored in a shorter vector that, after rehash, is only as long as the number of keys.  (`nk` here is the number of keys before rehash--i.e., it includes deleted keys.)

This might suggest that the strategy should be different.  An alternative could be, e.g. 

```suggestion
    if h.ndel >= ((3*nk)>>2) >= 3 || cnt*3 > sz*2
```

which basically requires at least 4 keys before anything happens.